relatio/narrative_models.py

```diff
@@ -344,23 +346,37 @@ def fit_static_clustering(
                                 args[k] = v
 
                         hdb = hdbscan.HDBSCAN(**args).fit(self.training_vectors)
-
                         models.append(hdb)
-
                         score = hdbscan.validity.validity_index(
                             self.training_vectors.astype(np.float64), hdb.labels_
                         )
                         scores.append(score)
-
+                        plot_args["min_cluster_size"].append(i)
+                        plot_args["min_samples"].append(j)
+                        plot_args["score"].append(score)
+
+            best_score_args = {}
+            max_index = np.argmax(scores)
+            best_score_args["min_cluster_size"] = plot_args["min_cluster_size"][
+                max_index
+            ]
+            best_score_args["min_samples"] = plot_args["min_samples"][max_index]
+            best_score_args["score"] = plot_args["score"][max_index]
+
+            self.plot_args = plot_args
+            self.best_score_args = best_score_args
             self.scores["DBCV"] = scores
-
             l = np.argmax(self.scores["DBCV"])
             print(
                 "The DBCV score suggests the index of the optimal clustering model is {0}.".format(
                     l
                 )
             )
 
+        self.clustering_model = models[np.argmax(scores)]
```

Comment:
self.clustering_model doesn't exist anymore and was replaced by self.clustering_models

Reply:
Lines 377 to 379 also appear redundant